### PR TITLE
Enable savepints for all engines. Add disable_savepoints param

### LINKED ISSUE #44: SQLite do supports SAVEPOINT
Following SQLAchemy's documentation, [savepoint with SQLite](https://docs.sqlalchemy.org/en/13/dialects/sqlite.html#savepoint-support) is supported (excluding some minro issues with the pysqlite driver where a workaround is avaible).

When trying to use SAVEPOINT with SQLite, the following error is raised:

```python
raceback (most recent call last):
  File "/lib/python3.7/site-packages/transaction/_transaction.py", line 631, in __init__
    savepoint = datamanager.savepoint
  File "/lib/python3.7/site-packages/zope/sqlalchemy/datamanager.py", line 155, in savepoint
    raise AttributeError("savepoint")
AttributeError: savepoint
```

This is due because SQLite is explicitly declared as not supported:

https://github.com/zopefoundation/zope.sqlalchemy/blob/ae4745cd85f5b04445929e8c1761e1a60e02a441/src/zope/sqlalchemy/datamanager.py#L59
https://github.com/zopefoundation/zope.sqlalchemy/blob/ae4745cd85f5b04445929e8c1761e1a60e02a441/src/zope/sqlalchemy/datamanager.py#L150-L155

However, if I remove ``sqlite`` from the set, my transaction is acting as expected:

```log
[2020-03-30 21:01:12 - INFO/sqlalchemy.engine.base.Engine:711] BEGIN (implicit)
[2020-03-30 21:01:12 - INFO/sqlalchemy.engine.base.Engine:1205] SELECT xxx
[2020-03-30 21:01:12 - INFO/sqlalchemy.engine.base.Engine:1210] ()
[2020-03-30 21:01:12 - INFO/sqlalchemy.engine.base.Engine:1205] SAVEPOINT sa_savepoint_1
[2020-03-30 21:01:12 - INFO/sqlalchemy.engine.base.Engine:1210] ()
[2020-03-30 21:01:12 - INFO/sqlalchemy.engine.base.Engine:1205] INSERT INTO xxx
[2020-03-30 21:01:12 - INFO/sqlalchemy.engine.base.Engine:1210] ()
[2020-03-30 21:01:12 - INFO/sqlalchemy.engine.base.Engine:1205] ROLLBACK TO SAVEPOINT sa_savepoint_1
[2020-03-30 21:01:12 - INFO/sqlalchemy.engine.base.Engine:1210] ()
[2020-03-30 21:01:12 - INFO/sqlalchemy.engine.base.Engine:1205] SELECT xxx
```

My environment:

```log
SQLAlchemy==1.3.15
transaction==3.0.0
zope.sqlalchemy==1.3
```

### PR DESCRIPTION
**Pull request by lrowe**
Fixes #44. Looking for feedback but requires documentation before merging.

From my Travis tests it appears Python 3.6 was the first to ship with a SQLite version that supports savepoints. (It's possible to hack in support using pysqlite for earlier versions using: https://github.com/lrowe/pysqlite/wiki/Building-pysqlite-with-static-env-and-savepoints.)

Should this be considered a breaking change? Potentially those running older versions of SQLite may now need to add `disable_savepoints=True` to their register call.

### CODE REVIEW

**Comment by icemac** [comment]
I like this PR although I am not using this library against SQLite on Python < 3.6.
I'd consider this a breaking change as we should not yet drop support for Python < 3.6.

File: src/zope/sqlalchemy/datamanager.py [hunk 1]
```diff
@@ -147,11 +146,7 @@ def savepoint(self):
         # support savepoints but Postgres is whitelisted independent of its
         # version. Possibly additional version information should be taken
         # into account (ajung)
```
Comment (mgedmin) [inline src/zope/sqlalchemy/datamanager.py]: This comment is outdated now you're no longer checking engine names.

**Comment by lrowe** [comment]
@mgedmin `transaction.savepoint()` without `optimistic=True` raises a TypeError when one of its data managers is missing a savepoint method. To get that same behaviour you would need to use `zope.sqlalchemy.register(disable_savepoints=True)`.

If not and your database/driver does not support subtransactions you get a `sqlalchemy.exc.OperationalError` when the database errors on the `transaction.commit()` or `.rollback()` having called `transaction.savepoint()` in the transaction. Example test failures before I added the version based unittest.skipIf: https://travis-ci.org/github/zopefoundation/zope.sqlalchemy/jobs/668966939

I'm reluctant to try and make a decision automagically as the logic required will likely be incomplete and error prone. "Explicit is better than implicit" and all that.

Python 2.7 and 3.5 sqlite who are using savepoints (likely optimistic ones) should either:
* stay on zope.sqlalchemy 1.3
* specify `disable_savepoints=True`
* install pysqlite >= 2.8.0 to get working savepoints

According to https://devguide.python.org/#status-of-python-branches Python 3.5 end-of-life  is September this year. Perhaps it's enough to just clearly document this and call it out in the changelog?

**Comment by icemac** [comment]
I think documenting is enough.

**Comment by dwt** [comment]
What exactly is missing to get this pull request to continue? I just had to add an ugly workaround in my app because of this - and I would like to get rid of it soon. :-)

**Comment by sallner** [comment]
@dwt We need the following:

- [ ] Rebase to master
- [ ] Add documentation of the non-support in readme and changelog
- [ ] Fix comment in code. 

@lrowe  Do you intend to work on that soon, or should someone else step in?

**Comment by icemac** [comment]
I am closing this old PR. Feel free to re-open it if you want to work on it.

**Pull request closed without merging**